apis/v1alpha1/tools/gen_policies.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"log"
+	"os"
+	"text/template"
+)
+
+//go:generate go run gen_policies.go
+
+// Define policy types
+var policies = []struct {
+	Name         string
+	HasSingleRef bool // true if it uses a single TargetRef, false if it uses TargetRefs
+}{
+	{"ClientSettingsPolicy", true},
+	{"ObservabilityPolicy", false},
+	{"UpstreamSettingsPolicy", false},
+}
+
+// Template for generating policy methods
+const tmpl = `// Code generated by gen_policies.go; DO NOT EDIT.
```

Comment:
Let's deduplicate the common template and main function that we have in both v1alpha1 and v1alpha2 into a common file, maybe in `apis/` top-level. Then we can just include the specific policy lists in each api version directory.

The package version at the top of the template could also be provided as a templated value.